chalice/deploy/planner.py

```diff
@@ -1227,7 +1227,7 @@ def _plan_restapi(self, resource):
                 (models.APICall(
                     method_name='import_rest_api',
                     params={'swagger_document': resource.swagger_doc,
-                            'endpoint_type': resource.endpoint_type},
+                            'endpoint': resource.endpoint},
```

Comment:
We should convert this down to the native python types.  `resource.endpoint` is a `model.Endpoint`, and right now the `awsclient` module knows nothing about `models`.

As a result this fails for me when trying to import the rest API:

```
$ chalice deploy
...
Creating Rest API
Traceback (most recent call last):
  File "chalice/chalice/cli/__init__.py", line 656, in main
    return cli(obj={})
  File ".virtualenvs/chalice-37/lib/python3.7/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File ".virtualenvs/chalice-37/lib/python3.7/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File ".virtualenvs/chalice-37/lib/python3.7/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File ".virtualenvs/chalice-37/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File ".virtualenvs/chalice-37/lib/python3.7/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File ".virtualenvs/chalice-37/lib/python3.7/site-packages/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "chalice/chalice/cli/__init__.py", line 216, in deploy
    deployed_values = d.deploy(config, chalice_stage_name=stage)
  File "chalice/chalice/deploy/deployer.py", line 376, in deploy
    return self._deploy(config, chalice_stage_name)
  File "chalice/chalice/deploy/deployer.py", line 392, in _deploy
    self._executor.execute(plan)
  File "chalice/chalice/deploy/executor.py", line 43, in execute
    self._default_handler)(instruction)
  File "chalice/chalice/deploy/executor.py", line 55, in _do_apicall
    result = method(**final_kwargs)
  File "chalice/chalice/awsclient.py", line 1069, in import_rest_api
    'endpointConfigurationTypes': endpoint['type']
TypeError: 'Endpoint' object is not subscriptable
```